### PR TITLE
Feat/combine gwas workflow attrition table logic Quantitative Only

### PR DESCRIPTION
**Pull request by jarvisraymond-uchicago**
Jira Ticket: [VADC-277](https://ctds-planx.atlassian.net/browse/VADC-277)

# Description
* Updates the Attrition table to support quantitative use case in new workflow
* Shows second table when outcome.variable_type === 'custom_dichotomous'
### New Features
* Adds unit test to check second table renders when outcome.variable_type === 'custom_dichotomous'
* Adds Storybook for new component Attrition Table wrapper

### Bugfixes
* Updates command eslint-new to continue even if deleted file is found in change list. 

### Implementation 
output

### CODE REVIEW

File: src/Analysis/GWASV2/GWASContainer.jsx [hunk 1]
```diff
@@ -15,41 +15,40 @@ const GWASContainer = () => {
     setSelectedStudyPopulationCohort,
   ] = useState({});
   const [selectedControlCohort] = useState(undefined);
-  const [selectedCaseCohort] = useState(undefined);
-  const [selectedCovariates] = useState([]);
-  const [selectedDichotomousCovariates] = useState([]);
+  const [newCovariateSubset] = useState([]);
+  const [outcome] = useState({});
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/GWASContainer.jsx:19]: I think I prefer what is used 2 lines above:
```
useState(undefined)
```
It makes the code more readable and avoids things like:
```
Object.keys(selectedStudyPopulationCohort).length === 0
```

File: src/Analysis/GWASV2/GWASContainer.jsx [hunk 2]
```diff
@@ -15,41 +15,40 @@ const GWASContainer = () => {
...
+        return null;
     }
   };
 
   let nextButtonEnabled = true;
   if (
-    current === 0
-    && Object.keys(selectedStudyPopulationCohort).length === 0
+    current === 0 &&
+    Object.keys(selectedStudyPopulationCohort).length === 0
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/GWASContainer.jsx:51]: see comment above.

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTable.jsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                <th className='gwasv2-attrition-table--w15'>Hispanic</th>
+              </tr>
+            </thead>
+            <tbody>
+              {selectedCohort?.cohort_definition_id && (
+                <React.Fragment key={selectedCohort}>
+                  {/* This is for the first Cohort Row in the Table */}
+                  <AttritionTableRow
+                    selectedCohort={selectedCohort}
+                    outcome={{}}
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTable.jsx:76]: use undefined instead

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx [hunk 4]
```diff
@@ -7,16 +6,16 @@ import {
...
-import BarChart from './ChartIcons/BarChart';
-import EulerDiagram from './ChartIcons/EulerDiagram';
-import './AttritionTable.css';
+} from '../../wizardEndpoints/cohortMiddlewareApi';
+import BarChart from '../ChartIcons/BarChart';
+import EulerDiagram from '../ChartIcons/EulerDiagram';
 
 const AttritionTableRow = ({
-  cohortDefinitionId,
   otherCohortDefinitionId,
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx:14]: let's drop `otherCohortDefinitionId`

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx [hunk 5]
```diff
@@ -7,16 +6,16 @@ import {
...
+  selectedCohort,
   rowType,
-  rowName,
+  rowObject,
+  outcome,
   covariateSubset,
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx:19]: the best name for this one is `currentCovariateAndCovariatesFromPrecedingRows`

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx [hunk 6]
```diff
@@ -80,65 +75,80 @@ const AttritionTableRow = ({
...
 
-  const determineChartIcon = (rowTypeInput) => {
-    if (rowTypeInput === 'Cohort') {
-      return null;
+  const determineChartIcon = () => {
+    if (rowType === 'Covariate' || rowType === 'Outcome') {
+      if (rowObject.variable_type === 'concept') {
+        return <BarChart />;
+      }
+      return <EulerDiagram />;
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx:98]: maybe it is best if this is an `else if (rowObject.variable_type === 'custom_dichotomous')`? And then add an `else` that will throw an error?

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx [hunk 7]
```diff
@@ -80,65 +75,80 @@ const AttritionTableRow = ({
...
+  rowObject: PropTypes.object.isRequired,
+  selectedCohort: PropTypes.object,
   sourceId: PropTypes.number.isRequired,
 };
 
 AttritionTableRow.defaultProps = {
-  cohortDefinitionId: undefined,
   otherCohortDefinitionId: undefined,
+  selectedCohort: undefined,
+  outcome: undefined,
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx:151]: yes 👍

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx [hunk 8]
```diff
@@ -80,65 +75,80 @@ const AttritionTableRow = ({
...
+  outcome: PropTypes.object,
+  rowObject: PropTypes.object.isRequired,
+  selectedCohort: PropTypes.object,
   sourceId: PropTypes.number.isRequired,
 };
 
 AttritionTableRow.defaultProps = {
-  cohortDefinitionId: undefined,
   otherCohortDefinitionId: undefined,
+  selectedCohort: undefined,
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx:150]: no, this one should be required

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTableWrapper.jsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  selectedCohort: PropTypes.object,
+  otherSelectedCohort: PropTypes.object,
+  outcome: PropTypes.object,
+  newCovariateSubset: PropTypes.array.isRequired,
+  sourceId: PropTypes.number.isRequired,
+};
+
+AttritionTableWrapper.defaultProps = {
+  selectedCohort: undefined,
+  otherSelectedCohort: undefined,
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTableWrapper.jsx:47]: let's drop `otherSelectedCohort`

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTableWrapper.jsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  <React.Fragment>
+    <AttritionTable
+      sourceId={sourceId}
+      newCovariateSubset={newCovariateSubset}
+      selectedCohort={selectedCohort}
+      otherSelectedCohort={otherSelectedCohort}
+      outcome={outcome}
+      // selectedCovariates={selectedCovariates}
+      // selectedDichotomousCovariates={selectedDichotomousCovariates}
+      tableHeader={'Case Cohort Attrition Table'}
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTableWrapper.jsx:21]: we should start with just "Attrition Table". 
In step 2, once the outcome is known, and _only if_ it is `custom_dichotomous`, then we do the following:
- set tableHeader of first table to `{'Case Cohort Attrition Table'}`
- set tableHeader of second table to `{'Control Cohort Attrition Table'}`

**Review by pieterlukasse — COMMENTED** [review]
thanks @jarvisraymond-uchicago ! I added some comments, mostly for cleaning up some unnecessary code and some ideas for better naming.

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx (hunk 5 above)
Comment (pieterlukasse) [reply src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx:19]: or `currentCovariatePlusCovariatesFromPrecedingRows` 🤔

Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx:19]: I prefer the first one, plus could imply addition

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTable.jsx [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+import PropTypes from 'prop-types';
+import { Collapse } from 'antd';
+import AttritionTableRow from './AttritionTableRow';
+import './AttritionTable.css';
+
+const { Panel } = Collapse;
+
+const AttritionTable = ({
+  selectedCohort,
+  otherSelectedCohort,
```
[the remark this answers is not in the document]
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTable.jsx:11]: Done. I also removed: 

```
otherCohortDefinitionId={
                          otherSelectedCohort
                            ? otherSelectedCohort.cohort_definition_id
                            : undefined
                        }

```

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx (hunk 4 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx:14]: This has been removed, including from the useQuery call in AttritionTableRow

```
 !otherCohortDefinitionId
        ? fetchConceptStatsByHareSubset(
            cohortDefinitionId,
            currentCovariateAndCovariatesFromPrecedingRows,
            outcome,
            sourceId
          )
        : fetchConceptStatsByHareSubsetCC(
```

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTableWrapper.jsx (hunk 9 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTableWrapper.jsx:43]: Removed.

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTableWrapper.jsx (hunk 10 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTableWrapper.jsx:19]: Interesting!

Updated to 

```
const useSecondTable = outcome.variable_type === 'custom_dichotomous';

```

And then for the first table

```
       tableHeader={
          useSecondTable ? 'Case Cohort Attrition Table' : 'Attrition Table'
        }
```

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx (hunk 4 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx]: Removed.

File: src/Analysis/GWASV2/GWASContainer.jsx (hunk 2 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/GWASContainer.jsx]: Updated to 

 if (current === 0 && !selectedStudyPopulationCohort) {

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx (hunk 7 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx]: Updated to not required and outcome: null,

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+
+const AttritionTableRow = ({
+  selectedCohort,
+  rowType,
+  rowObject,
+  outcome,
+  currentCovariateAndCovariatesFromPrecedingRows,
+  sourceId,
+}) => {
+  const [breakdownSize, setBreakdownSize] = useState(0);
```
Comment (jarvisraymond-uchicago) [inline src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx:21]: @pieterlukasse : FYI: These variables were initially declared as undefined, then boolean conditionals (lines 132-140) set them to zero if false. I updated the state to initialize them at zero and removed the conditional.

Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx]: Per discussion this morning, the declarations are now initialized to null

File: src/Analysis/GWASV2/GWASContainer.jsx [hunk 13]
```diff
@@ -13,11 +13,10 @@ const GWASContainer = () => {
   const [
     selectedStudyPopulationCohort,
     setSelectedStudyPopulationCohort,
-  ] = useState({});
-  const [selectedControlCohort] = useState(undefined);
-  const [selectedCaseCohort] = useState(undefined);
-  const [selectedCovariates] = useState([]);
-  const [selectedDichotomousCovariates] = useState([]);
+  ] = useState(null);
+  // const [selectedControlCohort] = useState({});
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/GWASContainer.jsx:17]: can we remove this?

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTable.jsx [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+            </thead>
+            <tbody>
+              {selectedCohort?.cohort_definition_id && (
+                <React.Fragment key={selectedCohort}>
+                  {/* This is for the first Cohort Row in the Table */}
+                  <AttritionTableRow
+                    selectedCohort={selectedCohort}
+                    outcome={null}
+                    rowType='Cohort'
+                    rowObject={{}}
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTable.jsx:74]: null?

**Review by pieterlukasse — APPROVED** [review]
looks good to me. Just two minor comments below

File: src/Analysis/GWASV2/GWASContainer.jsx (hunk 13 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/GWASContainer.jsx:17]: Removed.

File: src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTable.jsx (hunk 14 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTable.jsx]: Update to use null here and default to null as an unrequired prop in AttritionTableRow.jsx line 167.

**Pull request merged**